lib/chat_web/live/main_live/index.html.heex

```diff
@@ -297,6 +303,20 @@
           </button>
         </div>
       </.modal>
+      <.modal id="secret-share-popup" class="t-logout-without-key-modal">
+        <h1 class="text-base font-bold text-grayscale">Share a Secret</h1>
+        <p class="mt-3 text-sm text-black/50">
+          Select users who you want to share your secrets with.
+        </p>
+        <.live_component module={ChatWeb.MainLive.Layout.SecretSharingForm} id="secret" me={@me} />
+
+        <button
+          phx-click={hide_modal("secret-share-popup")}
+          class="w-full mr-1 h-12 border rounded-lg border-black/10"
+        >
+          Cancel
+        </button>
+      </.modal>
```

Comment:
You can also represent the entire modal as a LiveComponent. See `open_modal/3`. 
This is not necessary, but just FYI